miden-tx/src/tx_benches/utils.rs

```diff
@@ -0,0 +1,267 @@
+extern crate alloc;
+pub use alloc::{
+    string::{String, ToString},
+    vec::Vec,
+};
+
+use miden_lib::transaction::TransactionTrace;
+use miden_objects::{
+    accounts::{Account, AccountId, AccountStub},
+    assembly::ModuleAst,
+    notes::NoteId,
+    transaction::{ChainMmr, InputNote, InputNotes, TransactionArgs},
+    BlockHeader,
+};
+use miden_tx::{DataStore, DataStoreError, TransactionHost, TransactionInputs};
+use mock::mock::{
+    account::MockAccountType, notes::AssetPreservationStatus, transaction::mock_inputs,
+};
+use vm_processor::{
+    AdviceExtractor, AdviceInjector, AdviceProvider, ExecutionError, Host, HostResponse,
+    ProcessState,
+};
+
+// CONSTANTS
+// ================================================================================================
+
+/// Number of cycles needed to create an empty span whithout changing the stack state.
+const SPAN_CREATION_SHIFT: u32 = 2;
+
+// BENCHMARK HOST
+// ================================================================================================
+
+/// Wrapper around [TransactionHost] used for benchmarking: `tx_progress` allows to track the
+/// progression of the transaction execution.
+pub struct BenchHost<A: AdviceProvider> {
+    host: TransactionHost<A>,
+    tx_progress: TransactionProgress,
```

Comment:
Let's move `tx_progress` (and associated structs) in the main `TransactionHost`.